backend/ReservationApp/views.py

```diff
@@ -50,3 +52,24 @@ def post(self, request):
         send_confirmation_email(start_time, user.email, restaurant)
 
         return make_success_response()
+
+class ReservationUpdateStateView(APIView):
+    def put(self, request):
+        try:
+            restaurant_id = request.data['restaurant_id']
+            user_id = request.data['user_id']
+            new_state = request.data['new_state']
+        except KeyError as e:
+            return Response({'error': f'Missing required field: {e}'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
The format of an error response needs to be the same as it is for the other routes so that the frontend has an easier time rendering them. so they should be like:
```
errors: {
error_type_1: ['error message 1', 'error message 2'],
error_type_2: ['error message 3']
}
```

This is a pretty weird format, but the Django form validation follows it so we have to as well :(. You can try making a reservation with more than 30 people in postman for a better look at how it should be formatted.